torch/csrc/jit/codegen/cuda/scheduler/reduction_utils.cpp

```diff
@@ -396,145 +320,16 @@ void multiReductionInliner(
-        compute_to,
-        -1,
-        ComputeAtMode::BestEffort,
-        mapped_to_trivial_reduction);
+  }
 
-  } else {
-    // Want to inline, especially backwards based on reduction_tv, otherwise
-    // rfactor tv may not be inlined correctly
-    auto ref_tvs = rfactor_tvs.size() ? rfactor_tvs : reduction_tvs;
-    for (auto red_tv : ref_tvs) {
-      auto pos_it = std::find_if(
-          red_tv->domain()->domain().begin(),
-          red_tv->domain()->domain().end(),
-          [&mapped_to_trivial_reduction](IterDomain* id) {
-            return id->getParallelType() == ParallelType::Unswitch ||
-                id->getParallelType() == ParallelType::Unroll ||
-                id->getParallelType() == ParallelType::Vectorize ||
-                id->getParallelType() == ParallelType::MisalignedVectorize ||
-                mapped_to_trivial_reduction.count(id);
-          });
-      auto pos = pos_it == red_tv->domain()->domain().end()
-          ? -1
-          : std::distance(red_tv->domain()->domain().begin(), pos_it) + 1;
-
-      scheduler_utils::computeAtInputs(red_tv, pos, ComputeAtMode::MostInlined);
-      scheduler_utils::computeWithOutputs(
-          red_tv, pos, ComputeAtMode::BestEffort);
-    }
-    // For topologies where there may not be paths to all inputs/outputs from
-    // the reductions, we need to take a similar approach to the unrolled
-    // version and setup of compute at from inputs->outputs avoiding going
-    // through the reduction expressions. This can be done by grabbing inputs
-    // not on path to a reduction, and computeAt-ing with all outputs. This
-    // doesn't guarantee we don't go through a reduction, but with best effort
-    // it should minimize damage if it does.
-    std::vector<TensorView*> compute_to;
-    for (auto out : ir_utils::filterByType<TensorView>(fusion->outputs())) {
-      // only terminating outputs
-      if (out->uses().size() || out->isFusionInput()) {
-        continue;
-      }
-      compute_to.push_back(out);
-    }
+  // Inline the schedule
+  InlinePropagator inline_propagator(
+      reference_tv, -1, ComputeAtMode::MostInlined, {}, mapped_to_trivial_reduction);
 
-    std::vector<TensorView*> compute_from;
-    std::unordered_set<TensorView*> inps_of_reds;
-    {
-      auto inps_of_red_vec = ir_utils::inputTvsOf(ref_tvs);
-      inps_of_reds = std::unordered_set<TensorView*>(
-          inps_of_red_vec.begin(), inps_of_red_vec.end());
-    }
-    for (auto inp : ir_utils::filterByType<TensorView>(fusion->inputs())) {
-      if (inps_of_reds.find(inp) != inps_of_reds.end()) {
-        continue;
-      }
-      compute_from.push_back(inp);
-    }
+  MaxRootDomainInfoSpanningTree path(reference_tv);
 
-    scheduler_utils::computeAtBetween(
-        compute_from,
-        compute_to,
-        -1,
-        ComputeAtMode::BestEffort,
-        mapped_to_trivial_reduction);
-  }
+  path.traverse(&inline_propagator);
+  inline_propagator.tearDown();
```

Comment:
`tearDown` is automatically called by `path.traverse`, so you don't have to do it manually.

Reply:
Oh thanks, that was left as I was debugging and getting confused what was happening. I needed the expression sorting fix because we're now setting computeAt for output tensors, but expression sorting can just ignore it.